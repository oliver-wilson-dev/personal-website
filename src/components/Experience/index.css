.paragraphEmoji {
    margin-right: 10px;
    font-size: 30px;
}

.jobRole {
    display: inline-block;
    margin-bottom: 0;
    margin-right: 5px;
}

.jobRole + h4 {
    font-size: 16px;
    font-weight: 400;
    letter-spacing: 2px;
    margin-top:5px;
    text-transform: uppercase;
}

.companyTitle {
    margin-bottom: 0px;
    font-family: 'Source Sans Pro', sans-serif;
}

.projectImages {
    display: flex;
}

.projectImage {
    flex-basis: 50%;
    margin: 0px 10px 10px 0px;
}

.lastProjectImage {
    margin-right: 0;
}

.projectImage img {
    width: 100%;
    height: auto;
    box-shadow: 1px 1px 4px var(--primaryBorderColour);
}

@media (min-width: 786px) {
    .projectImages {
        flex-wrap: wrap;
    }

    .projectImage {
        flex-basis: 100%;
        margin: 10px 0px 0px 0px;
        padding: 10px 15px;
    }

    .projectImage img {
        box-shadow: 5px 5px 0px var(--primaryBorderColour);
        border: 3px solid var(--primaryBorderColour)
    }

    .projectImage:first-child {
        margin: 0;
    }
}


@media (min-width: 0px) and (max-width: 1023px) {
    .tilesFlowOffScreen {
        margin: 0px -30px;
    }
}
