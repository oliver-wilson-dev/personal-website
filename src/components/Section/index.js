import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import sharedStyles from '../App/index.css';
import styles from './index.css';
import useFadeInClasses from '../../hooks/useFadeInClasses';

const Section = ({ title, children, additionalStyles = {} }) => {
  const { fadeInClasses } = useFadeInClasses();

  return (
    <section className={cn(
      styles.flexColumn,
      sharedStyles.flexCenter,
      fadeInClasses,
      { [additionalStyles.section]: !!additionalStyles.section, }
    )}
    >
      {title && <h2 className={styles.sectionTitle}>{title}</h2>}
      <div className={additionalStyles.container}>
        {children}
      </div>
    </section>
  );
};

Section.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node.isRequired,
  additionalStyles: PropTypes.oneOfType([
    PropTypes.shape()
  ])
};

export default Section;
