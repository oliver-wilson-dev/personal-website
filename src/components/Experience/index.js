import React from 'react';
import PropTypes from 'prop-types';
import Section from '../Section';
import Workplace from '../Workplace';
import Slider from '../Slider';
import SectionTile from '../SectionTile';
import styles from './index.css';

const Experience = ({ title, workplaces }) => (
  <Section title={title} additionalStyles={{ section: styles.tilesFlowOffScreen }}>
    <Slider>
      {workplaces.map(({ companyName, ...rest }) => (
        <SectionTile key={companyName}>
          <Workplace
            {...{ ...rest, companyName }}
          />
        </SectionTile>
      ))}
    </Slider>
  </Section>
);

Experience.propTypes = {
  title: PropTypes.string.isRequired,
  workplaces: PropTypes.array.isRequired
};

export default Experience;
