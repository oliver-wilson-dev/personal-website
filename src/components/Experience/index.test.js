import React from 'react';
import { shallow } from 'enzyme';
import Experience from '.';
import Section from '../Section';

const defaultProps = {
  title: 'test-title',
  emoji: 'test-emoji',
  workplaces: [
    { companyName: 'test-company-name' }, { companyName: 'test-another-company-name' }
  ]
};

jest.mock('../Workplace', () => {
  const Workplace = () => null;

  return Workplace;
});


jest.mock('../Section', () => {
  const Section = () => null;

  return Section;
});

jest.mock('../SectionTile', () => {
  const SectionTile = () => null;

  return SectionTile;
});

const render = overrideProps => shallow(<Experience {...defaultProps} {...overrideProps} />);

describe('<Experience/> component', () => {
  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should match snapshot', () => {
    expect(render()).toMatchSnapshot();
  });

  it('should render a Section component and pass it the "title" prop', () => {
    const testTitle = 'test-title';
    expect(render({ title: testTitle }).find(Section).prop('title')).toBe(testTitle);
  });
});
