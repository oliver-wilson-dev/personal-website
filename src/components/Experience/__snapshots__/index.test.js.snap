// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Experience/> component should match snapshot 1`] = `
<Section
  additionalStyles={
    Object {
      "section": "tilesFlowOffScreen",
    }
  }
  title="test-title"
>
  <Slider>
    <SectionTile
      key="test-company-name"
    >
      <Workplace
        companyName="test-company-name"
      />
    </SectionTile>
    <SectionTile
      key="test-another-company-name"
    >
      <Workplace
        companyName="test-another-company-name"
      />
    </SectionTile>
  </Slider>
</Section>
`;
